src/init.rs

```diff
@@ -0,0 +1,40 @@
+pub fn init(shell_name: &str) {
+    log::debug!("Shell name: {}", shell_name);
+    let setup_script = match shell_name {
+        "bash" => {
+            let script = "PS1=\"$(starship prompt --status=$?)\"";
+            Some(script)
+        }
+        "zsh" => {
+            let script = "PROMPT=\"$(starship prompt --status=$?)\"";
+            Some(script)
+        }
+        "fish" => {
+            let script = "function fish_prompt; starship prompt --status=$status; end";
+            Some(script)
+        }
+        _ => {
+            println!(
+                "printf \"\\n{0} is not yet supported by starship.\\n\
```

Comment:
Is there supposed to be a `printf` inside the `println!`?

Reply:
Yes. What is within that `println()` is wrapped in an `eval()`, so the printf is needed to have it print the newlines as expected.

Reply:
Understand now @matchai!